testsuite/MDAnalysisTests/analysis/test_wbridge.py

```diff
@@ -202,34 +636,552 @@ def test_count_by_type_multiple_frame(self):
+        }
+
+    kwargs = {
+        'selection1': 'protein',
+        'selection2': 'protein',
+        'detect_hydrogens': "distance",
+        'distance': 3.0,
+        'angle': 150.0,
+    }
+
+    @pytest.fixture(scope='class')
+    def h(self, universe):
+        kw = self.kwargs.copy()
+        # kw.update(kwargs)
+        h = MDAnalysis.analysis.hbonds.WaterBridgeAnalysis(universe, order=0, **kw)
+        # remove in 1.0
+        if kw['detect_hydrogens'] == 'heuristic':
+            with pytest.warns(DeprecationWarning):
+                h.run(verbose=False)
+        else:
+            h.run(verbose=False)
+        return h
+
+    def test_helix_backbone(self, values, h):
+        assert len(h.timeseries[0]) == values['num_bb_hbonds'], "wrong number of backbone hydrogen bonds"
+        assert h.timesteps, [0.0]
+
+    def test_generate_table(self, values, h):
+
+        h.generate_table()
+        assert len(h.table) == values['num_bb_hbonds'], "wrong number of backbone hydrogen bonds in table"
+        assert isinstance(h.table, np.core.records.recarray)
+        h.table.sort(order='donor_resid')
+        assert_array_equal(h.table.donor_resid, values['donor_resid'])
+        assert_array_equal(h.table.acceptor_resnm, values['acceptor_resnm'])
+
+    def test_atoms_too_far(self):
+        pdb = '''
+ATOM      1  N   LEU     1      32.310  13.778  14.372  1.00  0.00      SYST N 0
+ATOM      2  OW  SOL     2       3.024   4.456   4.147  1.00  0.00      SYST H 0'''
+
+        u = MDAnalysis.Universe(StringIO(pdb), format="pdb")
+        h = WaterBridgeAnalysis(u, 'resname SOL', 'protein', order=0)
+        h.run(verbose=False)
+        assert h.timeseries == [[]]
+
+    def test_acceptor_OC1_OC2(self):
+        gro = '''test
+3
+    1ALA    OC1    1   0.000   0.000   0.000
+    2ALA      N    2   0.300   0.000   0.000
+    2ALA     H1    3   0.200   0.000   0.000
+7.29748 7.66094 9.82962'''
+
+        u = MDAnalysis.Universe(StringIO(gro), format="gro")
+        h = WaterBridgeAnalysis(u, 'protein', 'protein', order=0)
+        h.run(verbose=False)
+        assert h.timeseries[0][0][2] == 'ALA2:H1'
+
+    def test_true_traj(self):
+        u = MDAnalysis.Universe(GRO, XTC)
+        u.add_TopologyAttr(guess_types(u.atoms.names))
+        h = WaterBridgeAnalysis(u, 'protein', 'resname ASP', distance=3.0, angle=120.0, order=0)
+        h.run()
+        assert len(h.timeseries) == 10
+
+    def test_count_by_time(self, values, h):
+
+        c = h.count_by_time()
+        assert c, [(0.0, values['num_bb_hbonds'])]
+
+    def test_count_by_type(self, values, h):
+
+        c = h.count_by_type()
+        assert_equal([line[-1] for line in c][:8], values['num_bb_hbonds'] * [1.0])
+
+    def test_timesteps_by_type(self, values, h):
+
+        t = h.timesteps_by_type()
+        assert_equal([i[-1] for i in t][:8], values['num_bb_hbonds'] * [0.0])
+
+
+class TestHydrogenBondAnalysisPBC(TestHydrogenBondAnalysis):
```

Comment:
name?

Reply:
I originally add this part so that I can be sure that the wba passed all the hba test. Now as the test suite for the wba is rewritten, I have removed the hba test.